colossalai/inference/tensor_parallel/engine.py

```diff
@@ -36,17 +57,52 @@ class TPInferEngine:
         >>> generate_kwargs = ...
         >>> shard_config = ShardConfig(enable_tensor_parallelism=True, inference_only=True)
         >>> infer_engine = TPInferEngine(model, shard_config, MAX_BATCH_SIZE, MAX_INPUT_LEN, MAX_OUTPUT_LEN)
+        >>> infer_engine.optimize_model()
         >>> outputs = infer_engine.generate(input_ids, **generate_kwargs)
     """
 
     def __init__(self,
-                 model: nn.Module,
-                 shard_config: ShardConfig,
-                 max_batch_size: int,
-                 max_input_len: int,
-                 max_output_len: int,
+                 model: Union[str, nn.Module],
+                 shard_config: ShardConfig = None,
+                 max_batch_size: int = 1,
+                 max_input_len: int = 1,
+                 max_output_len: int = 1,
```

Comment:
Might want to modify these default values

Reply:
Done